R/fb_league_stats.R

```diff
@@ -0,0 +1,150 @@
+#' @importFrom purrr map_dfr
+#' @importFrom dplyr mutate
+#' @importFrom readr type_convert
+.fb_single_league_stats <- function(url, team_or_player) {
+
+  session <- worldfootballr_chromote_session(url)
+  tables <- worldfootballr_html_table(session)
+
+  idx <- if(team_or_player == "player") { 3 } else { 1:2 }
+
+  purrr::map_dfr(
+    idx,
+    ~{
+      table <- tables[[.x]]
+
+      renamed_table <- .rename_fb_cols(table)
+
+      renamed_table <- if (team_or_player == "player") {
+        renamed_table[renamed_table$Rk != "Rk", ]
+      } else {
+        renamed_table %>%
+          dplyr::mutate(
+            "Team_or_Opponent" = ifelse(.x == 1, "team", "opponent"),
+            .before = 1
+          )
+
+      }
+
+      suppressMessages(
+        readr::type_convert(
+          renamed_table,
+          guess_integer = TRUE,
+          na = "",
+          trim_ws = TRUE
+        )
+      )
+    }
+  )
+
+}
+
+#' Get FBref Team or Player Season Statistics for an Entire League
+#'
+#' Get the season stats for all teams / players in a selected league
+#'
+#' @inheritParams fb_match_urls
+#' @param stat_type the type of statistic required. Must be one of the following:
+#' \itemize{
+#' \item{standard}
+#' \item{keepers}
+#' \item{keepersadv}
+#' \item{shooting}
+#' \item{passing}
+#' \item{passing_types}
+#' \item{gca}
+#' \item{defense}
+#' \item{possession}
+#' \item{playing_time}
+#' \item{misc}
+#' }
+#'
+#' @return a dataframe of season stats for all teams / players in a league
+#'
+#' @importFrom rlang arg_match0 .data .env
+#' @importFrom magrittr %>%
+#' @importFrom dplyr filter pull transmute
+#' @importFrom tibble tibble
+#' @importFrom tidyr crossing
+#' @importFrom purrr possibly map_dfr
+#' @importFrom stats setNames
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' try({
+#' fb_season_team_stats(
+#'   country = "ENG",
+#'   gender = "M",
+#'   season_end_year = 2022,
+#'   tier = "1st",
+#'   stat_type = "shooting",
+#'   team_or_player = "player"
+#' )
+#' })
+#' }
+fb_league_stats <- function(country, gender, season_end_year, tier = "1st", non_dom_league_url = NA, stat_type, team_or_player) {
```

Comment:
note that in some ways this function is similar to `fb_player_season_stats`, but also has some elements from `fb_match_urls()` (the first part where we retrieve league URLs)

Reply:
Yeah lot's of similarities, only now we can grab player data. Awesome